### ISSUE TITLE
i3-autolayout is adding extra nodes

### ISSUE THREAD

**Issue by joehillen** [BiagioFesta/i3-autolayout#10]
I've been noticing that when I try to switch to tabmode after opening a new window nothing happens. I have to move the new window up in the i3 tree and then toggle tabmode. This is because i3-autolayout is adding an extra node around each new node.

Here is a repo case:

1. start i3-autolayout
2. open a new workspace
3. open a terminal
4. open another terminal
5. Run `i3-msg -t get_tree | jq '.. | objects | select(.type == "workspace" and .num == 5)'` (replace 5 with your workspace number)
6. See the number of nodes (add `| .. | .id? | numbers` to the above command) **I get 5**
7. kill i3-autolayout
8. repeat 2-6 **I get 3**

I'll dig into the code later, but I figured I should tell you before I do in case you have a quicker fix.

**Comment by BiagioFesta** [comment]
Thanks for reporting this, and sorry for my late reply. 

I've broken my laptop lately, and I am struggling to keep up with daily personal activities. Sorry about that.

Feel free to dig into it, I will have a look as soon as I'll be back operative :)

**Comment by BiagioFesta** [comment]
Hi,
I was able to get back on your issue.

---

*TL;DR;, I think this is an expected behavior*

---

By following your description, I don't think this is really an issue; but rather how i3 layout data structure works.
Lemme try to explain:

If I totally disable `i3-autolayout`:
 * Be sure it is disabled: `systemctl --user stop i3-autolayout`
 * Go on workspace `X`
 * Open 3 terminal.
   * Because no autolayout I will obtain 3 windows side by side on the horizontal split.
```
┌───────────────────────────────────────────┐
│                                           │
│ ┌───────────┐ ┌───────────┐ ┌───────────┐ │
│ │           │ │           │ │           │ │
│ │           │ │           │ │           │ │
│ │  w1       │ │   w2      │ │    w3     │ │
│ │           │ │           │ │           │ │
│ │           │ │           │ │           │ │
│ │           │ │           │ │           │ │
│ └───────────┘ └───────────┘ └───────────┘ │
│                                           │
└───────────────────────────────────────────┘
```
 * New version of `i3-autolayout` allows printing the i3 tree structure (it should be more simple than parsing `i3-msg`).
   * `i3-autolayout print-tree X`  // where `X` is the i3-workspace number.
```
[ID: 94280865186592; Type: Workspace; Name: Some("4"); Layout: SplitH; WinType: None; NumFloatings: 0]
├─ [ID: 94280865093776; Type: Con; Name: Some("Alacritty"); Layout: SplitH; WinType: Some(Normal); NumFloatings: 0]
├─ [ID: 94280865092528; Type: Con; Name: Some("Alacritty"); Layout: SplitH; WinType: Some(Normal); NumFloatings: 0]
└─ [ID: 94280865336176; Type: Con; Name: Some("Alacritty"); Layout: SplitH; WinType: Some(Normal); NumFloatings: 0]
```

So I have a root-node (`Type Workspace`) and 3 children in this "container" (3 terminals; `Alacritty` is my terminal emulator).

However, if you want to perform some different split with `i3`, you need to create "other containers" with a different layout.

For example, (*always keeping `i3-autolayout` disabled*). Try to create 3 terminals with the following configuration:
```

  ┌──────────────────────────────┐
  │                              │
  │ ┌───────────┐ ┌───────────┐  │
  │ │           │ │   2       │  │
  │ │           │ │           │  │
  │ │   1       │ ├───────────┤  │
  │ │           │ ├───────────┤  │
  │ │           │ │           │  │
  │ │           │ │    3      │  │
  │ └───────────┘ └───────────┘  │
  │                              │
  │                              │
  └──────────────────────────────┘
```

The keystrokes should be something like
  * `meta + enter` (create first terminal)
  * `meta + enter` (create second terminal)
  * `meta + v` (focusing the second terminal to apply a vertical split)
  * `meta + enter` (create third terminal)

Now check the i3 tree structure:
```
[ID: 94280865186592; Type: Workspace; Name: Some("4"); Layout: SplitH; WinType: None; NumFloatings: 0]
├─ [ID: 94280865092528; Type: Con; Name: Some("Alacritty"); Layout: SplitH; WinType: Some(Normal); NumFloatings: 0]
└─ [ID: 94280865337232; Type: Con; Name: None; Layout: SplitV; WinType: None; NumFloatings: 0]
   ├─ [ID: 94280865093776; Type: Con; Name: Some("Alacritty"); Layout: SplitH; WinType: Some(Normal); NumFloatings: 0]
   └─ [ID: 94280865343728; Type: Con; Name: Some("Alacritty"); Layout: SplitH; WinType: Some(Normal); NumFloatings: 0]
```

You will see that `i3` creates a new node, a container (`94280865337232`, `WinType: None;`). You can think this like a window container containing the vertical split.

---

In conclusion, arranging windows with different layouts implies creation of new "empty" node/container for different layout split (despite of the usage of `i3-autolayout`).

Let me know if I am missing something in your issue.
